From 8d117d7c628c43ba0bcaa84fa7b3fef085c39e12 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Tue, 13 Nov 2018 20:18:41 -0700
Subject: [PATCH 11/15] msm8974-common: sepolicy: Resolve mediaserver denials

* avc: denied { read } for name="/" dev="tmpfs" ino=6954
  scontext=u:r:mediaserver:s0 tcontext=u:object_r:device:s0
  tclass=dir permissive=1
* avc: denied { open } for name="/" dev="tmpfs" ino=6954
  scontext=u:r:mediaserver:s0 tcontext=u:object_r:device:s0
  tclass=dir permissive=1
* avc: denied { search } for name="graphics" dev="sysfs" ino=12879
  scontext=u:r:mediaserver:s0 tcontext=u:object_r:sysfs_graphics:s0
  tclass=dir permissive=0

Change-Id: I32540b8ec48ff61568094ec87d1bad64570a8e55
---
 sepolicy/common/mediaserver.te | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sepolicy/common/mediaserver.te b/sepolicy/common/mediaserver.te
index 7990af5..c696f84 100644
--- a/sepolicy/common/mediaserver.te
+++ b/sepolicy/common/mediaserver.te
@@ -1,7 +1,9 @@
 allow mediaserver camera_socket:sock_file write;
+allow mediaserver device:dir r_dir_perms;
 allow mediaserver hal_camera_hwservice:hwservice_manager find;
 allow mediaserver mm-qcamerad:unix_dgram_socket sendto;
 allow mediaserver sysfs_camera:dir search;
 allow mediaserver sysfs_camera:file r_file_perms;
+allow mediaserver sysfs_graphics:dir search;
 allow mediaserver thermal-engine:unix_stream_socket connectto;
 allow mediaserver vendor_file:file execmod;
-- 
2.17.1

